### ISSUE TITLE
a11y: Use current filter background color with stronger contrast

### ISSUE THREAD

**Issue by fkling** [sourcegraph/sourcegraph#36534]

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.

**Comment by benvenker** [comment]
@almeidapaulooliveira can you connect with @fkling in your morning and see what needs to be done here?

**Comment by almeidapaulooliveira** [comment]
@fkling do you have more context? For example, the link for the code to know if it is the light or the dark theme? Also, it would help to understand the criteria that flagged as low contrast. Is it some numeric hue difference?

**Comment by fkling** [comment]
@almeidapaulooliveira See https://buildkite.com/sourcegraph/sourcegraph/builds/149563#0180fa2b-6e3c-4d94-9bdc-502dbd89ceb8/227-313 The contrast between the highlight color and the text color doesn't meet the criteria.

```
Rule: "color-contrast" (Elements must have sufficient color contrast)
--
  | Further information: https://dequeuniversity.com/rules/axe/4.4/color-contrast?application=axe-puppeteer
  | How to manually audit: https://docs.sourcegraph.com/dev/background-information/web/accessibility/how-to-audit#auditing-a-user-journey
  | Required changes:
  | <span class="CodeMirrorQueryInput-module__field_24sEh">-file</span>:
  | Fix any of the following:
  | Element has insufficient color contrast of 4.02 (foreground color: #0b70db, background color: #e6ebf2, font size: 9.0pt (12px), font weight: normal). Expected contrast ratio of 4.5:1
  | 
```

**Comment by almeidapaulooliveira** [comment]
@fkling can you confirm this rule is complaining about the sidebar panel? I don't see this class used on it.

<img width="1016" alt="Screen Shot 2022-07-19 at 11 48 23" src="https://user-images.githubusercontent.com/64257673/179780190-058b68de-face-494f-9f08-5d9124f3f41a.png">

**Comment by almeidapaulooliveira** [comment]
@benvenker, @fkling, I figured out where the issue is. I updated the [Figma file](https://www.figma.com/file/8MkIcpDtjv6tqgxZFcEQDt/Core-workflow-punch-list?node-id=762%3A23830) with the updated colors with correct contrast on the Light theme (checked it with the [suggested tool](https://dequeuniversity.com/rules/axe/4.4/color-contrast?application=axe-puppeteer)), and the Dark theme doesn't appear to have a problem.

I also would like to ask a favor. With issues like that would be nice to receive a clear problem definition (pointing to the code or element that the contrast failed) instead of an unclear solution (remove or "better" color).

**Comment by fkling** [comment]
@almeidapaulooliveira This issue is about the background color of the "current" filter in the search query input:
<img width="502" alt="2022-07-19_18-25" src="https://user-images.githubusercontent.com/179026/179801283-253a84d2-8082-4672-851e-245c2aacb76a.png">

The color is set here: https://sourcegraph.sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/search-ui/src/input/CodeMirrorQueryInput.module.scss?L181-190

I don't know how fixable this is though given the fact that we also apply syntax highlighting to the query itself (thus different characters will have different colors and thus different foreground/background color contrast).

**Comment by benvenker** [comment]
cc @taylorsperry do you have any opinions on a good solution here? We're not sure of the best path forward.

**Comment by taylorsperry** [comment]
@umpox is this related to the work that you and @rrhyne did?

**Comment by fkling** [comment]
Are we going to do anything about this or not?

**Comment by felixfbecker** [comment]
@almeidapaulooliveira could you provide a different text or background color so that the contrast is at least 4.5:1 here?

**Comment by rrhyne** [comment]
@almeidapaulooliveira lets get this issue resolved!

**Comment by almeidapaulooliveira** [comment]
We need to adjust the blue color globally to avoid inconsistencies in the App.

## Light theme:
Changing the `--blue` color from `#0b70db` to `#0B6AD0` would work.

It's a slight difference...
![colors-comparision](https://user-images.githubusercontent.com/64257673/202183463-a4459774-a421-47f6-808a-6f8605dc1abf.png)

...but enough to pass the test:
![Screen Shot 2022-11-16 at 09 32 39](https://user-images.githubusercontent.com/64257673/202183498-5babe8a5-6324-4f16-b9c1-1dff7626ec6a.png)

## Dark theme: 
  Changing the `--search-filter-keyword-color` color from `#4393e7` to `#4BA2F6` would work.
 
   But the dark theme isn’t using a global blue, so I would recommend adjusting other parts of the App for consistency. For example, `--link-color` (`#4393e7`) that could be `#4BA2F6`. But a further investigation it's required.
      
 @quinnkeast, could you please review this change?
